docs/WalletGuide.md

```diff
@@ -49,7 +54,52 @@ There is one more available configuration for wallet that allows to configure in
 For example, to test with local servers on http protocol, http can be manually enabled.
 
 ```kotlin
-val walletCustom = Wallet(StellarConfiguration.Testnet, ApplicationConfiguration(useHttp = true))
+val walletCustom = Wallet(
+  StellarConfiguration.Testnet,
+  ApplicationConfiguration { defaultRequest { url { protocol = URLProtocol.HTTP } } }
+)
+```
+
+### Configuring client
+Wallet SDK uses [ktor client](https://ktor.io/docs/getting-started-ktor-client.html) for all network requests 
+(excluding Horizon, where SDK http client is used). Currently, okhttp engine is configured to be used with the client.  
+Ktor has a reach client configuration, you can read about it more [here](https://ktor.io/docs/create-client.html#configure-client).
+For example, client can be globally configured:
+```kotlin
+val walletCustomClient =
+  Wallet(
+    StellarConfiguration.Testnet,
+    ApplicationConfiguration(
+      defaultClientConfig = {
+        engine { this.config { this.connectTimeout(Duration.ofSeconds(10)) } }
+        install(HttpRequestRetry) {
+          retryOnServerErrors(maxRetries = 5)
+          exponentialDelay()
+        }
+      }
+    )
+  )
+```
+This code will set connect timeout to 10 seconds via 
+[okhttp configuration](https://ktor.io/docs/http-client-engines.html#okhttp) 
+and install [retry plugin](https://ktor.io/docs/client-retry.html)  
+Some services also allows client configuration. For example, to change connect timeout when connecting to recovery 
+servers:
+```kotlin
+val recoveryCustomClient =
+  walletCustomClient.recovery {
+    engine { this.config { this.connectTimeout(Duration.ofSeconds(30)) } }
+  }
+```
+
+### Closing resources
+After wallet class is no longer used, it's necessary to close all clients used by it. While in some applications it may 
+not be required (e.g. wallet lives for the whole lifetime of the app), in other cases it can be required.  
+If your wallet class is short-lived it's recommended to close client resources using close function:
```

Comment:
```suggestion
After the wallet class is no longer used, it's necessary to close all clients used by it. While in some applications it may 
not be required (e.g. the wallet lives for the whole lifetime of the app), in other cases it can be required.  
If your wallet class is short-lived, it's recommended to close client resources using close function:
```